### PR TITLE
Reupload: Added numpy bf16 datatype support via custom pip package

### PR DESCRIPTION
**Pull request by jakpiase**
### PR types
Others

### PR changes
Others

### Describe
This PR adds bfloat16 datatype support for numpy via an external package. Sample usage is included in UT. That will shorten the time of loading word2vec BF16 model significantly and will allow us to remove "convert_float_to_uint16" function from UTs. It is a reupload of #35332

### CODE REVIEW

**Comment by lidanqing-vv** [comment]
@Aganlengzi  Hi this is the new PR of bf16 numpy package. Please help review. Thanks !

File: python/requirements.txt [hunk 1]
```diff
@@ -1,7 +1,9 @@
 requests>=2.20.0
-numpy>=1.13
+numpy>=1.13 ; python_version>="3.5" and platform_system != "Windows"
+numpy>=1.13, <=1.19.3 ; python_version>="3.5" and platform_system == "Windows"
```
Comment (Aganlengzi) [inline python/requirements.txt]: Plz notice that the numpy version changes on develop branch are reverted, is this needed?

Comment (jakpiase) [reply python/requirements.txt]: No, it's just a copy-paste error, thank you for your awareness

**Review by Aganlengzi — COMMENTED** [review]
LGTM

**Comment by Aganlengzi** [comment]
Info about this package from #35332 @phlrain @jzhang533 @dingjiaweiww @kolinwei 
Additional info:
My PIP package: https://pypi.org/project/paddle-bfloat/
Github source code(changes are on develop2 branch): https://github.com/jakpiase/paddle_bfloat/tree/develop2

This package is an extension to https://pypi.org/project/bfloat16/ package. My package added python2 and windows compatibility and made it possible to install it with older pip versions. Original(https://pypi.org/project/bfloat16/) package is an extension to Tensorflow's bfloat16 numpy implementation which made it possible to use as a standalone feature. The author have written a note in bfloat16.cc file about his changes in Tensorflow's files(to make it clear that it's still Tensorflow's license, but he did some changes to source code) and so did I. I haven't changed any licensing and it is distributed on Apache2(same as Tensorflow). I had to make a separate pip package because some PaddlePaddle machines have old pip versions and it was not possible for pip to install package directly from github or from local files.

**Review by jzhang533 — APPROVED** [review]
Discussed with SongJiaChen offline, LGTM.

**Review by kolinwei — APPROVED** [review]

**Review by phlrain — APPROVED** [review]

**Review by dingjiaweiww — APPROVED** [review]

**Comment by lidanqing-vv** [comment]
@jakpiase  Hi, we need to compile locally, on three different platforms (Linux, Windows, MacOS), to get .whl package for three different platforms and upload those .whl to pypi

You need to have a PyPi account to upload to its library. Upload the dist/*.whl file on PyPi by using Twine:
```
python -m twine upload dist/*
```

**Pull request merged**